src/lib/DataConnectStore.js

```diff
@@ -59,6 +59,22 @@ export default class DataConnectStore {
     return useCase.connectors.map(c1 => this.find(c2 => c1.slug === c2.slug))
   }
 
+  // Fetch all accounts and updates their matching connectors
+  fetchAllAccounts (index) {
+    if (!index && this.accountsIndex) index = this.accountsIndex
```

Comment:
We could have many index on the same documents for example, not here for data-connect but let's see for photos. We could use different index (by Name, by date or by other meta property) with the only one same listing fetch function, couldn't it?